python-package/xgboost/core.py

```diff
@@ -2203,96 +2205,43 @@ def get_score(self, fmap='', importance_type='weight'):
-                    arr = line.split('[')
-                    # if no opening bracket (leaf node), ignore this line
-                    if len(arr) == 1:
-                        continue
-
-                    # extract feature name from string between []
-                    fid = arr[1].split(']')[0].split('<')[0]
-
-                    if fid not in fmap:
-                        # if the feature hasn't been seen yet
-                        fmap[fid] = 1
-                    else:
-                        fmap[fid] += 1
-
-            return fmap
-
-        average_over_splits = True
-        if importance_type == 'total_gain':
-            importance_type = 'gain'
-            average_over_splits = False
-        elif importance_type == 'total_cover':
-            importance_type = 'cover'
-            average_over_splits = False
-
-        trees = self.get_dump(fmap, with_stats=True)
-
-        importance_type += '='
-        fmap = {}
-        gmap = {}
-        for tree in trees:
-            for line in tree.split('\n'):
-                # look for the opening square bracket
-                arr = line.split('[')
-                # if no opening bracket (leaf node), ignore this line
-                if len(arr) == 1:
-                    continue
-
-                # look for the closing bracket, extract only info within that bracket
-                fid = arr[1].split(']')
-
-                # extract gain or cover from string after closing bracket
-                g = float(fid[1].split(importance_type)[1].split(',')[0])
-
-                # extract feature name from string before closing bracket
-                fid = fid[0].split('<')[0]
-
-                if fid not in fmap:
-                    # if the feature hasn't been seen yet
-                    fmap[fid] = 1
-                    gmap[fid] = g
-                else:
-                    fmap[fid] += 1
-                    gmap[fid] += g
-
-        # calculate average value (gain/cover) for each feature
-        if average_over_splits:
-            for fid in gmap:
-                gmap[fid] = gmap[fid] / fmap[fid]
-
-        return gmap
+        args = from_pystr_to_cstr(
+            json.dumps({"importance_type": importance_type, "feature_map": fmap})
+        )
+        features = ctypes.POINTER(ctypes.c_char_p)()
+        scores = ctypes.POINTER(ctypes.c_float)()
+        length = c_bst_ulong()
+        _check_call(
+            _LIB.XGBoosterFeatureScore(
+                self.handle,
+                args,
+                ctypes.byref(length),
+                ctypes.byref(features),
+                ctypes.byref(scores)
+            )
+        )
+        features_arr = from_cstr_to_pystr(features, length)
+        scores_arr = ctypes2numpy(scores, length.value, np.float32)
+        results = {}
+        for feat, score in zip(features_arr, scores_arr):
+            results[feat] = score
+        return results
```

Comment:
Hi @trivialfis, can we convert `score` to the native python float type here for a use case below?

```python
model  = xgb.train(...)
imp = model. get_score(...)

with open(filepath, "w") as f:
    json.dump(imp, f)  # this line throws if `score` is of `numpy.float32`
```

Reply:
Ofcourse, we can do `imp = {k: float(v) for k, v in imp}`

Reply:
Thanks for pointing that out, I will update with a new PR.

Reply:
thanks for the quick action!